### ISSUE TITLE
MacOS v11 + Bun v1.0.20 + Vips v8.14.5 + Sharp v0.32.6 = fatal error vips/vips8 file not found

### ISSUE THREAD

**Issue by baptistecs** [lovell/sharp#3927]
> I guess this would work fine with Vips v8.15 and Sharp v0.33, but it seems I can't install Vips v8.15 on MacOS v11 (with Homebrew requiring a more recent MacOS version or MacPorts only providing Vips v8.14), and vips build instructions indicate to use Homebrew (catch 22).
> If there is a way to build vips without Homebrew, I guess there would be no issue?
> I apologize if this is the wrong place to ask for help.

## Possible install-time or require-time problem

- [x] I have read and understood all of the [documentation relating to installation](https://sharp.pixelplumbing.com/install).
- [x] I have searched for known bugs relating to this problem in my choice of package manager.

You must confirm both of these before continuing.

### Are you using the latest version of sharp?

- [ ] I am using the latest version of `sharp` as reported by `npm view sharp dist-tags.latest`.

If you cannot confirm this, please upgrade to the latest version and try again before opening an issue.

If you are using another package which depends on a version of `sharp` that is not the latest,
please open an issue against that package instead.

> I can't use the last version because I use an old MacBook Pro (Mid 2014) which is up to date with the last supported MacOS version available (Big Sur v11.7.10), Homebrew won't install Vips version ^8.15 as it require a most recent version of MacOS, Mac Ports allow to install Vips v8.14.5, then I need to downgrade Sharp to v0.32.6 which generate a fatal error.

There is homebrew output:
<details>
➜  work brew install vips
==> Downloading https://formulae.brew.sh/api/cask.jws.json 100.0%
Warning: You are using macOS 11.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
</details>

### Are you using a supported runtime?

- [ ] I am using Node.js with a version that satisfies `^18.17.0 || ^20.3.0 || >=21.0.0`
- [ ] I am using Deno
- [x] I am using Bun

If you cannot confirm any of these,
please upgrade to the latest version
and try again before opening an issue.

### Are you using a supported package manager and installing optional dependencies?

- [ ] I am using npm >= 9.6.5 with `--include=optional`
- [ ] I am using yarn >= 3.2.0
- [ ] I am using pnpm >= 7.1.0 with `--no-optional=false`
- [ ] I am using Deno
- [x] I am using Bun

If you cannot confirm any of these, please upgrade to the latest version of your chosen package manager
and ensure you are allowing the installation of optional or multi-platform dependencies before opening an issue.

### What is the complete output of running `npm install --verbose --foreground-scripts sharp` in an empty directory?

Same error:
<details>

npm install --verbose --foreground-scripts sharp@^0.32 23:45:05 2/01/24
npm verb cli /Users/UN/.nvm/versions/node/v18.16.1/bin/node /Users/UN/.nvm/versions/node/v18.16.1/bin/npm
npm info using npm@9.5.1
npm info using node@v18.16.1
npm verb title npm install sharp@^0.32
npm verb argv "install" "--loglevel" "verbose" "--foreground-scripts" "sharp@^0.32"
npm verb logfile logs-max:10 dir:/Users/UN/.npm/_logs/2024-01-02T22_45_17_283Z-
npm verb logfile /Users/UN/.npm/_logs/2024-01-02T22_45_17_283Z-debug-0.log
npm http fetch GET 200 https://registry.npmjs.org/sharp 16ms (cache hit)
npm http fetch GET 200 https://registry.npmjs.org/detect-libc 78ms (cache hit)
npm http fetch GET 200 https://registry.npmjs.org/color 80ms (cache hit)
npm http fetch GET 200 https://registry.npmjs.org/semver 79ms (cache hit)
npm http fetch GET 200 https://registry.npmjs.org/tunnel-agent 140ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/tar-fs 143ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/simple-get 146ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/node-addon-api 150ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/prebuild-install 150ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/github-from-package 55ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/mkdirp-classic 58ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/color-convert 62ms (cache hit)
npm http fetch GET 200 https://registry.npmjs.org/color-string 62ms (cache hit)
npm http fetch GET 200 https://registry.npmjs.org/pump 59ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/lru-cache 60ms (cache hit)
npm http fetch GET 200 https://registry.npmjs.org/expand-template 69ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/minimist 69ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/decompress-response 87ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/once 87ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/napi-build-utils 91ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/simple-concat 89ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/rc 92ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/safe-buffer 86ms (cache updated)
npm http fetch GET 200 https://registry.npmjs.org/mkdirp-classic 91ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/tar-stream 90ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/node-abi 101ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/color-name 10ms (cache hit)
npm http fetch GET 200 https://registry.npmjs.org/simple-swizzle 9ms (cache hit)
npm http fetch GET 200 https://registry.npmjs.org/color-name 12ms (cache hit)
npm http fetch GET 200 https://registry.npmjs.org/is-arrayish 3ms (cache hit)
npm http fetch GET 200 https://registry.npmjs.org/chownr 47ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/end-of-stream 51ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/deep-extend 50ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/strip-json-comments 51ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/ini 54ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/wrappy 116ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/yallist 4ms (cache hit)
npm http fetch GET 200 https://registry.npmjs.org/mimic-response 28ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/fast-fifo 33ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/b4a 35ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/streamx 63ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/queue-tick 731ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/inherits 33ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/fs-constants 35ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/bl 37ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/readable-stream 39ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/string_decoder 33ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/buffer 35ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/util-deprecate 37ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/base64-js 33ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/ieee754 33ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/string_decoder/-/string_decoder-1.3.0.tgz 253ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/base64-js/-/base64-js-1.5.1.tgz 252ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/util-deprecate/-/util-deprecate-1.0.2.tgz 258ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/fs-constants/-/fs-constants-1.0.0.tgz 245ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/inherits/-/inherits-2.0.4.tgz 253ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/queue-tick/-/queue-tick-1.0.1.tgz 250ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/chownr/-/chownr-1.1.4.tgz 252ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/simple-concat/-/simple-concat-1.0.1.tgz 245ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/ieee754/-/ieee754-1.2.1.tgz 263ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/deep-extend/-/deep-extend-0.6.0.tgz 249ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/tar-stream/-/tar-stream-3.1.6.tgz 261ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/mimic-response/-/mimic-response-3.1.0.tgz 260ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/strip-json-comments/-/strip-json-comments-2.0.1.tgz 256ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/end-of-stream/-/end-of-stream-1.4.4.tgz 262ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/wrappy/-/wrappy-1.0.2.tgz 261ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/mkdirp-classic/-/mkdirp-classic-0.5.3.tgz 254ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/buffer/-/buffer-5.7.1.tgz 292ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/b4a/-/b4a-1.6.4.tgz 279ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/ini/-/ini-1.3.8.tgz 276ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/once/-/once-1.4.0.tgz 266ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/pump/-/pump-3.0.0.tgz 269ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/fast-fifo/-/fast-fifo-1.3.2.tgz 282ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/github-from-package/-/github-from-package-0.0.0.tgz 259ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/node-abi/-/node-abi-3.52.0.tgz 264ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/napi-build-utils/-/napi-build-utils-1.0.2.tgz 269ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/expand-template/-/expand-template-2.0.3.tgz 263ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/decompress-response/-/decompress-response-6.0.0.tgz 288ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/simple-get/-/simple-get-4.0.1.tgz 264ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/rc/-/rc-1.2.8.tgz 289ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/streamx/-/streamx-2.15.6.tgz 299ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/tar-fs/-/tar-fs-3.0.4.tgz 271ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/tunnel-agent/-/tunnel-agent-0.6.0.tgz 317ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/prebuild-install/-/prebuild-install-7.1.1.tgz 327ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/tar-stream/-/tar-stream-2.2.0.tgz 334ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/bl/-/bl-4.1.0.tgz 373ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/minimist/-/minimist-1.2.8.tgz 359ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/readable-stream/-/readable-stream-3.6.2.tgz 429ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/tar-fs/-/tar-fs-2.1.1.tgz 388ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/node-addon-api/-/node-addon-api-6.1.0.tgz 400ms (cache miss)
npm http fetch GET 200 https://registry.npmjs.org/sharp/-/sharp-0.32.6.tgz 413ms (cache miss)
npm info run sharp@0.32.6 install node_modules/sharp (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)

> sharp@0.32.6 install
> (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)

sharp: Detected globally-installed libvips v8.14.5@0.32.6 install node_modules/sharp (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
sharp: Building from source via node-gyp
gyp info it worked if it ends with oknfo run sharp@0.32.6 install node_modules/sharp (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
gyp verb cli [
gyp verb cli   '/Users/UN/.nvm/versions/node/v18.16.1/bin/node',
gyp verb cli   '/Users/UN/.nvm/versions/node/v18.16.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js',
gyp verb cli   'rebuild'
gyp verb cli ]
gyp info using node-gyp@9.3.1
gyp info using node@18.16.1 | darwin | x64
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directoryrun sharp@0.32.6 install node_modules/sharp (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
gyp verb command configure []
gyp verb find Python Python is not set from command line or npm configuration
gyp verb find Python Python is not set from environment variable PYTHON
gyp verb find Python checking if "python3" can be used
gyp verb find Python - executing "python3" to get executable path
gyp verb find Python - executable path is "/Applications/Xcode.app/Contents/Developer/usr/bin/python3"vips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
gyp verb find Python - executing "/Applications/Xcode.app/Contents/Developer/usr/bin/python3" to get version
gyp verb find Python - version is "3.8.2"run sharp@0.32.6 install node_modules/sharp (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
gyp info find Python using Python version 3.8.2 found at "/Applications/Xcode.app/Contents/Developer/usr/bin/python3"
gyp verb get node dir no --target version specified, falling back to host node version: 18.16.1
gyp verb command install [ '18.16.1' ]
gyp verb install input version string "18.16.1"arp@0.32.6 install node_modules/sharp (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
gyp verb install installing version: 18.16.1
gyp verb install --ensure was passed, so won't reinstall if already installed
gyp verb install version is already installed, need to check "installVersion"s/sharp (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
gyp verb got "installVersion" 11
gyp verb needs "installVersion" 9
gyp verb install version is good
gyp verb get node dir target node version installed: 18.16.1
gyp verb build dir attempting to create "build" dir: /Users/UN/work/testSharp/node_modules/sharp/build
gyp verb build dir "build" dir needed to be created? Yes
gyp verb python symlink creating symlink to "/Applications/Xcode.app/Contents/Developer/usr/bin/python3" at "/Users/UN/work/testSharp/node_modules/sharp/build/node_gyp_bins/python3"
gyp verb build/config.gypi creating config file
gyp verb build/config.gypi writing out config file: /Users/UN/work/testSharp/node_modules/sharp/build/config.gypi
gyp verb config.gypi checking for gypi file: /Users/UN/work/testSharp/node_modules/sharp/config.gypi
gyp verb common.gypi checking for gypi file: /Users/UN/work/testSharp/node_modules/sharp/common.gypi
gyp verb gyp gyp format was not specified; forcing "make"
gyp info spawn /Applications/Xcode.app/Contents/Developer/usr/bin/python3
gyp info spawn args [
gyp info spawn args   '/Users/UN/.nvm/versions/node/v18.16.1/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/UN/work/testSharp/node_modules/sharp/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/UN/.nvm/versions/node/v18.16.1/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/UN/Library/Caches/node-gyp/18.16.1/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/UN/Library/Caches/node-gyp/18.16.1',
gyp info spawn args   '-Dnode_gyp_dir=/Users/UN/.nvm/versions/node/v18.16.1/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/UN/Library/Caches/node-gyp/18.16.1/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/UN/work/testSharp/node_modules/sharp',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
Package hdf5 was not found in the pkg-config search path. install node_modules/sharp (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
Perhaps you should add the directory containing `hdf5.pc'
to the PKG_CONFIG_PATH environment variable
Package 'hdf5', required by 'matio', not found
gyp verb command build []ify:sharp: info run sharp@0.32.6 install node_modules/sharp (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
gyp verb build type Release
gyp verb architecture x64
gyp verb node dev dir /Users/UN/Library/Caches/node-gyp/18.16.1
gyp verb `which` succeeded for `make` /usr/bin/make
gyp verb bin symlinks adding symlinks (such as Python), at "/Users/UN/work/testSharp/node_modules/sharp/build/node_gyp_bins", to PATH
gyp info spawn make
gyp info spawn args [ 'V=1', 'BUILDTYPE=Release', '-C', 'build' ]
  cc -o Release/obj.target/nothing/../node-addon-api/nothing.o ../../node-addon-api/nothing.c '-DNODE_GYP_MODULE_NAME=nothing' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_GLIBCXX_USE_CXX11_ABI=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' -I/Users/UN/Library/Caches/node-gyp/18.16.1/include/node -I/Users/UN/Library/Caches/node-gyp/18.16.1/src -I/Users/UN/Library/Caches/node-gyp/18.16.1/deps/openssl/config -I/Users/UN/Library/Caches/node-gyp/18.16.1/deps/openssl/openssl/include -I/Users/UN/Library/Caches/node-gyp/18.16.1/deps/uv/include -I/Users/UN/Library/Caches/node-gyp/18.16.1/deps/zlib -I/Users/UN/Library/Caches/node-gyp/18.16.1/deps/v8/include  -O3 -gdwarf-2 -mmacosx-version-min=10.15 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -fno-strict-aliasing -MMD -MF ./Release/.deps/Release/obj.target/nothing/../node-addon-api/nothing.o.d.raw   -c
  rm -f Release/nothing.a && ./gyp-mac-tool filter-libtool libtool  -static -o Release/nothing.a Release/obj.target/nothing/../node-addon-api/nothing.o) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
warning: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: archive library: Release/nothing.a the table of contents is empty (no object file members in the library define global symbols)
  touch Release/obj.target/libvips-cpp.stamp
  c++ -o Release/obj.target/sharp-darwin-x64/src/common.o ../src/common.cc '-DNODE_GYP_MODULE_NAME=sharp-darwin-x64' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_GLIBCXX_USE_CXX11_ABI=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' '-DNAPI_VERSION=7' '-DNODE_ADDON_API_DISABLE_DEPRECATED' '-DNODE_API_SWALLOW_UNTHROWABLE_EXCEPTIONS' '-DBUILDING_NODE_EXTENSION' -I/Users/UN/Library/Caches/node-gyp/18.16.1/include/node -I/Users/UN/Library/Caches/node-gyp/18.16.1/src -I/Users/UN/Library/Caches/node-gyp/18.16.1/deps/openssl/config -I/Users/UN/Library/Caches/node-gyp/18.16.1/deps/openssl/openssl/include -I/Users/UN/Library/Caches/node-gyp/18.16.1/deps/uv/include -I/Users/UN/Library/Caches/node-gyp/18.16.1/deps/zlib -I/Users/UN/Library/Caches/node-gyp/18.16.1/deps/v8/include -I../../node-addon-api  -O3 -gdwarf-2 -mmacosx-version-min=10.13 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=c++11 -stdlib=libc++ -fexceptions -Wall -Oz -MMD -MF ./Release/.deps/Release/obj.target/sharp-darwin-x64/src/common.o.d.raw   -c
../src/common.cc:13:10: fatal error: 'vips/vips8' file not foundl node_modules/sharp (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
#include <vips/vips8>
         ^~~~~~~~~~~~
1 error generated.
make: *** [Release/obj.target/sharp-darwin-x64/src/common.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/UN/.nvm/versions/node/v18.16.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:203:23)
gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
gyp ERR! System Darwin 20.6.0
gyp ERR! command "/Users/UN/.nvm/versions/node/v18.16.1/bin/node" "/Users/UN/.nvm/versions/node/v18.16.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/UN/work/testSharp/node_modules/sharp
gyp ERR! node -v v18.16.1
gyp ERR! node-gyp -v v9.3.1
gyp ERR! not ok 
npm info run sharp@0.32.6 install { code: 1, signal: null }
npm verb stack Error: command failed
npm verb stack     at ChildProcess.<anonymous> (/Users/UN/.nvm/versions/node/v18.16.1/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:53:27)
npm verb stack     at ChildProcess.emit (node:events:513:28)
npm verb stack     at maybeClose (node:internal/child_process:1091:16)
npm verb stack     at ChildProcess._handle.onexit (node:internal/child_process:302:5)
npm verb pkgid sharp@0.32.6
npm verb cwd /Users/UN/work/testSharp
npm verb Darwin 20.6.0
npm verb node v18.16.1
npm verb npm  v9.5.1
npm ERR! code 1
npm ERR! path /Users/UN/work/testSharp/node_modules/sharp
npm ERR! command failed
npm ERR! command sh -c (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
npm verb exit 1
npm verb unfinished npm timer reify 1704235517338
npm verb unfinished npm timer reify:build 1704235519361
npm verb unfinished npm timer build 1704235519362
npm verb unfinished npm timer build:deps 1704235519362
npm verb unfinished npm timer build:run:install 1704235519368
npm verb unfinished npm timer build:run:install:node_modules/sharp 1704235519368
npm verb code 1

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/UN/.npm/_logs/2024-01-02T22_45_17_283Z-debug-0.log

</details>

### What is the output of running `npx envinfo --binaries --system --npmPackages=sharp --npmGlobalPackages=sharp`?

  System:
    OS: macOS 11.7.10
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
    Memory: 55.21 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 18.16.1 - ~/.nvm/versions/node/v18.16.1/bin/node
    Yarn: 3.6.1 - ~/.nvm/versions/node/v18.16.1/bin/yarn
    npm: 9.5.1 - ~/.nvm/versions/node/v18.16.1/bin/npm
    pnpm: 8.13.1 - ~/.nvm/versions/node/v18.16.1/bin/pnpm
    bun: 1.0.20 - ~/.bun/bin/bun

**Comment by baptistecs** [comment]
I added this in package.json:
```json
"trustedDependencies": ["sharp"],
```

And optionally this:
```json
  "resolutions": {
    "minipass": "2.7.0"
  },
```

And I tried:
```shell
rm -rf node_modules
rm -rf bun.lockb
bun install --no-cache --verbose --platform=darwin --arch=x64
```
But I get the same result 😫

**Comment by lovell** [comment]
Support for Bun was added in sharp v0.33.0.

> I can't use the last version because... Homebrew won't install Vips version ^8.15

You don't need a globally-installed libvips - use `brew uninstall vips` then upgrade sharp.

**Comment by baptistecs** [comment]
> Support for Bun was added in sharp v0.33.0.

Thanks a lot for the quick answer! :)

> You don't need a globally-installed libvips - use `brew uninstall vips` then upgrade sharp.

I think I need libvips to convert .heic files? So far I get this:

```node
Read error: source: bad seek to 3024940
heif: Unsupported feature: Unsupported codec (4.3000)
                                                            ^
TypeError: undefined is not an object (evaluating 'error.stack')
      at /Users/UN/work/PN/src/server.ts:186:145
      at processTicksAndRejections (:61:77)
```

**Comment by lovell** [comment]
The prebuilt binaries provided by sharp do not support HEVC/HEIC images.

MacPorts provides vips 8.15.1 https://ports.macports.org/port/vips/

**Comment by baptistecs** [comment]
Published 6h ago, great timing!

The installation went fine, but unfortunately, I can't test it ATM because I broke something with Bun, when I run `bun install` I get `error: failed to initialize compression stream downloading package manifest` for all the project packages (it worked fine before I started to play with HomeBrew / MacPorts / Vips build.

I close this issue for now, I will reopen if needed, thank you! :)
